--- Makefile	2013-09-25 14:16:18.000000000 +0200
+++ Makefile_patched	2019-10-02 11:55:09.267621597 +0200
@@ -59,7 +59,7 @@
 SOMINOR=1
 SORELEASE?=.0#   Declare empty to leave out from library file name.
 
-MINISAT_CXXFLAGS = -I. -D __STDC_LIMIT_MACROS -D __STDC_FORMAT_MACROS -Wall -Wno-parentheses -Wextra
+MINISAT_CXXFLAGS = -I. -D __STDC_LIMIT_MACROS -D __STDC_FORMAT_MACROS -Wall -Wno-parentheses -Wextra -fpermissive
 MINISAT_LDFLAGS  = -Wall -lz
 
 ECHO=@echo
